CONTRIBUTING.md

```diff
@@ -0,0 +1,183 @@
+# Contributing to Recast and Detour
+
+We'd love for you to contribute to our source code and to make Recast and Detour even better than they are
+today! Here are the guidelines we'd like you to follow:
+
+ - [Code of Conduct](#coc)
+ - [Question or Problem?](#question)
+ - [Issues and Bugs](#issue)
+ - [Feature Requests](#feature)
+ - [Submission Guidelines](#submission-guidelines)
+ - [Git Commit Guidelines](#git-commit-guidelines)
+
+## <a name="coc"></a> Code of Conduct
+This project adheres to the [Open Code of Conduct][code-of-conduct].
+By participating, you are expected to honor this code.
+
+## <a name="question"></a> Got a Question or Problem?
+
+If you have questions about how to use Recast or Detour, please direct these to the [Google Group][groups]
+discussion list. We are also available on [Gitter][gitter].
+
+## <a name="issue"></a> Found an Issue?
+If you find a bug in the source code or a mistake in the documentation, you can help us by
+submitting an issue to our [GitHub Repository][github]. Even better you can submit a Pull Request
+with a fix.
+
+**Please see the Submission Guidelines below**.
+
+## <a name="feature"></a> Want a Feature?
+You can request a new feature by submitting an issue to our [GitHub Repository][github].  If you
+would like to implement a new feature then consider what kind of change it is:
+
+* **Major Changes** that you wish to contribute to the project should be discussed first on our
+[Google Group][groups] or in [GitHub Issues][github-issues] so that we can better coordinate our efforts, prevent
+duplication of work, and help you to craft the change so that it is successfully accepted into the
+project.
+* **Small Changes** can be crafted and submitted to the [GitHub Repository][github] as a Pull Request.
+
+## Submission Guidelines
+
+### Submitting an Issue
+Before you submit your issue search the [GitHub Issues][github-issues] archive,
+maybe your question was already answered.
+
+If your issue appears to be a bug, and hasn't been reported, open a new issue.
+Help us to maximize the effort we can spend fixing issues and adding new
+features, by not reporting duplicate issues.  Providing the following information will increase the
+chances of your issue being dealt with quickly:
+
+* **Overview of the Issue** - what type of issue is it, and why is it an issue for you?
+* **Callstack** - if it's a crash or other runtime error, a callstack will help diagnosis
+* **Logs** - stdout and stderr from the console, or log files if there are any
+* **Reproduction steps** - a minimal, unambigious set of steps including input, that causes the error for you.
+    e.g. input geometry and settings you can use to input into RecastDemo to get it to fail.

```

Comment:
These can be saved by pressing the 9 key in RecastDemo, and the resulting .gset file can be shared (with the .obj if it is not one of the default ones).